CMakeLists.txt

```
@@ -869,6 +879,11 @@ if(openPMD_BUILD_EXAMPLES)
         if(${examplename} MATCHES ".+parallel$")
             if(openPMD_HAVE_MPI)
                 add_executable(${examplename} examples/${examplename}.cpp)
+                if (openPMD_HAVE_CUDA)
+                  enable_language(CUDA)
+                  target_link_libraries(${examplename} PUBLIC CUDA::cudart CUDA::cuda_driver)
+                  set_target_properties(${examplename} PROPERTIES CUDA_SEPARABLE_COMPILATION ON)
```

A few more things that we need:
```suggestion
                  set_source_files_properties(examples/${examplename}.cpp PROPERTIES
                    LANGUAGE CUDA )
                  target_compile_features(${examplename} PUBLIC cuda_std_17)
                  set_target_properties(${examplename} PROPERTIES
                    CUDA_SEPARABLE_COMPILATION ON
                    CUDA_EXTENSIONS OFF
                    CUDA_STANDARD_REQUIRED ON
                  )
```

and please but the `openpmd_cxx_required` in an `else` branch.

Also: is `CUDA_SEPARABLE_COMPILATION` really needed (e.g., does ADIOS2 already build with it and can you try if it works without it)?
I am asking, because separable compilation can lead to less efficient device code and is harder on the compiler toolchain (read as: often buggy).
In case separable compilation does not yet work, revisit if you use global (device) variables in your example and try to refactor them out.

---

I got linking errors when CUDA_SEPARABLE_COMPILATION is off, like this: 
/tmp/tmpxft_00159479_00000000-6_8a_benchmark_write_parallel.cudafe1.stub.c:14: undefined reference to `__cudaRegisterLinkedBinary_62_tmpxft_00159479_00000000_7_8a_benchmark_write_parallel_cpp1_ii_d7f2f159'

---

other suggestions are incorporated in the latest commit